provider/pkg/internal/pulumiapi/agent_pools.go

```diff
@@ -106,18 +107,23 @@ func (c *Client) UpdateAgentPool(ctx context.Context, agentPoolId, orgName, name
 	return nil
 }
 
-func (c *Client) DeleteAgentPool(ctx context.Context, agentPoolId, orgName string) error {
+func (c *Client) DeleteAgentPool(ctx context.Context, agentPoolId, orgName string, forceDestroy bool) error {
```

Comment:
super nit: should we consider making options like this a struct in case we need to add more later?